File: Python/Week 1/activity2.py
#!/usr/bin/python

import csv

DAYS = 365


def main():
    # specify path to temperature csv
    filePath = "Python\Week 1\\temperatures_365_days.csv"

    # initialise all dates slots for year
    dates = [None] * DAYS

    # initialise all temperature slots for year
    temperatures = [0] * DAYS

    readTemperatureData(filePath, dates, temperatures)
    findExtremeDates(dates, temperatures)

    print("Hottest day: {} with temperature {}°C".format(maxDay, maxTemp))
    print("Coldest day: {} with temperature {}°C".format(minDay, minTemp))


def readTemperatureData(filePath, dates, temperatures):
    with open(filePath) as csvDataFile:
        reader = csv.reader(csvDataFile)
        next(reader)
        i = 0
        for row in reader:
            dates[i] = row[0]
            temperatures[i] = row[1]
            i = i + 1
    csvDataFile.close


def findExtremeDates(dates, temperatures):
    global maxTemp
    maxTemp = temperatures[0]
    global minTemp
    minTemp = temperatures[0]
    global maxDay
    maxDay = dates[0]
    global minDay
    minDay = dates[0]

    for i in range(1, DAYS-1):
        if temperatures[i] > maxTemp:
            maxTemp = temperatures[i]
            maxDay = dates[i]
        if temperatures[i] < minTemp:
            minTemp = temperatures[i]
            minDay = dates[i]
    print("Hottest day: {} with temperature {}°C".format(maxDay, maxTemp))
    print("Coldest day: {} with temperature {}°C".format(minDay, minTemp))


if __name__ == '__main__':
    main()

File: Java/Week 1/activity_2.java
import java.io.File;
import java.io.FileNotFoundException;
import java.util.Scanner;


public class activity_2 {

    public static void main(String[] args) {
        
        String filePath = ("C:\\Users\\jarre\\Documents\\COMP1002\\Week 1\\temperatures_365_days.csv");
        
        String[] dates;
        dates = new String[365];
        
        int[] temperatures;
        temperatures = new int[365] ;

        readTemperatureData(filePath, dates, temperatures);
        findExtremeDays(dates, temperatures);



    }

    public static void readTemperatureData(String filePath, String[] dates, int[] temperatures) {
        
        try {
            File tempFile = new File(filePath);
            Scanner reader = new Scanner(tempFile);


            String[] buffer;
            buffer = new String[2];
            int j = 0;

            String information = reader.nextLine();

            while (reader.hasNextLine()){
                information = reader.nextLine();
                buffer = information.split(",");
                dates[j] = buffer[0];
                temperatures[j] = Integer.parseInt(buffer[1]);
                j++;
            }
            reader.close();
        } catch (FileNotFoundException e){
            System.out.println("Error, File Not Found!\n");
        }
    }

    public static void findExtremeDays(String[] dates, int[] temperatures) {
        int maxTemp = temperatures[0];
        int minTemp = temperatures[0];

        String maxDay = dates[0];
        String minDay = dates[0];

        int i;

        for (i = 0; i < 364; i++) {
            if (temperatures[i] > maxTemp)
                maxTemp = temperatures[i];
                maxDay = dates[i];
            
            if (temperatures[i] < minTemp)
                minTemp = temperatures[i];
                minDay = dates[i];
            
        }

        System.out.println("Hottest Day: " + maxDay + " with temperature " + maxTemp + "\u00B0C");
        System.out.println("Coldest Day: " + minDay + " with temperature " + minTemp + "\u00B0C");
    }
}
